Sometimes invalid `--filter` is passed for `dotnet test` in CircleCI

<https://app.circleci.com/pipelines/github/planetarium/libplanet/43/workflows/dcc3bef6-e020-4964-a26c-3460e111add0/jobs/234/parallel-runs/3?filterBy=FAILED>

The `dotnet test --list-tests` command apparently place more than one test case per line due to its bug:

```
Libplanet.Tests.AddressTest.Compare
Libplanet.Tests.HashcashTest.AnswerSatisfiesDifficulty(challenge: [112, 210, 160, 182, 210, ...], difficulty: 1048576)    Libplanet.Explorer.UnitTests.GraphTypes.AddressTypeTest.Serialize
Libplanet.Tests.Blocks.BlockHashTest.FromHashDigest
```

Such output causes an invalid `--filter` string for `dotnet test`, because `sed` script incorrectly parses the output:

          dotnet test --no-restore --no-build -c Release --list-tests \
          > .dotnet-list-tests.txt
          grep -E '^    ' .dotnet-list-tests.txt \
          | sed -E 's/^    |\(.*?\)$//g' \
          | uniq \
          | /usr/bin/sort -R --random-source=CHANGES.md \
          > "<<parameters.collect_tests_to>>"

Until .NET Core SDK's such I/O bug on Linux/macOS is fixed, we should work around it by making the `sed` script more flexible.

Additional debugging information: sometimes when two test cases occur in the same line, if the first case includes comma inside parentheses, this problem occurs: `MSBUILD : error MSB1006: Property is not valid.` This is because `dotnet test --filter` does not escape comma properly, and sed command inside CI script is supposed to strip the parentheses part but fails to do so since the line with two test cases do not match the regex.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -91,7 +91,10 @@ commands:
           fi
           dotnet test --no-restore --no-build -c Release --list-tests \
           > .dotnet-list-tests.txt
-          grep -E '^    ' .dotnet-list-tests.txt \
+          grep '    ' .dotnet-list-tests.txt \
+          | sed 's/    /\n    /g' \
+          | sed '/^$/d' \
+          | grep '^    ' \
           | sed -E 's/^    |\(.*?\)$//g' \
           | uniq \
           | /usr/bin/sort -R --random-source=CHANGES.md \

